### PR TITLE
Create update.yaml and .json

### PR DESCRIPTION
**Pull request by Saviq**
The purpose of this file is for Multipass to fetch it when checking for updates. It has more data than we can put in GitHub and it allows us to gauge usage.

Included is a .json file for compatibility with our [current code](https://github.com/CanonicalLtd/multipass/blob/20d04c0a770056a90abb0b77c55c6e457003d755/src/platform/update/new_release_monitor.cpp#L69) thats' parsing [GitHub's API](https://api.github.com/repos/CanonicalLtd/multipass/releases/latest).

### CODE REVIEW

File: update.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version: 1.0.0
+title: New Stable release
```
Comment (nottrobin) [inline update.yaml]: @Saviq are you planning to update this to real information before we merge it?

Comment (Saviq) [reply update.yaml]: @nottrobin yes, provided there's agreement on the format and approach :)

Comment (nottrobin) [reply update.yaml]: I think as a first pass the format is fine, and @anthonydillon appears to agree based on his approval, so please go ahead and make this real info and we can merge whenever you're ready.

Comment (Saviq) [reply update.yaml]: @nottrobin done.

File: update.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "html_url": "https://github.com/CanonicalLtd/multipass/releases/tag/v0.9.0",
```
Comment (nottrobin) [inline update.json]: Why do you need both `json` and `yaml`? Surely the multipass tool only needs to consume one of them. I'm not opinionated about which format you use, but both seems unnecessary.

Comment (nottrobin) [reply update.json]: (Although, if you're interested in my opinion, I would suggest `json` is the better format if it's intended purely for the `multipass` tool to consume)

**Comment by nottrobin** [comment]
@Saviq an important point @anthonydillon raised is - what will happen multipass side if https://multipass.run is down or slow? It would be a shame for it to degrade user experience in the multipass tool. We don't run this website as an API service currently, so our uptime guarantees etc. and not tailored towards supporting software infrastructure. I believe our uptime won't be too bad, but you should be very careful about how you implement this in multipass to fail gracefully, with a short timeout.

File: update.json (hunk 2 above)
Comment (ricab) [reply update.json]: The json is only meant to help us transition. If all goes well, it will be complete very soon and the json can be removed.

**Comment by nottrobin** [comment]
I think as long as you're okay with us not making strong promises about uptime or performance, this solution is okay as a stop-gap measure to solve your immediate need. So once you've responded about the 2-file question & the uptime question, we're probably happy to merge this.

However, I really don't think this is the ideal solution, and finding a more robust solution could be useful beyond multipass. I would suggest that we likely have many software tools with the same need - to inform users on the CLI when an update is available, and to gather metrics.

I would suggest that that should be based around an actual release / package management platform that natively has knowledge about the latest release. The information you've added here looks like it could be precisely provided  by the GitHub releases API, but a better solution might be to develop this capability through the Snap Store API, which presumably also has similar information?

**Comment by Saviq** [comment]
Hi @nottrobin as @ricab mentioned, the JSON is a backup in case we won't make the transition to YAML in time.

Re: uptime, yes we'll make sure that any trouble with the connection will not influence the result badly.

As for the long-term solution, I'm all ears, and agree that we seem to be producing more and more non-Linux solutions, so a more generic approach would be good to work out.

**Comment by nottrobin** [comment]
Okay then I think this LGTM for now. I'll see if I can reach out to @noise about a possible Snap Store based solution.

**Comment by nottrobin** [comment]
@Saviq, @ricab: actually, could I suggest that `json` is a better format for this?

Firstly, because it's intended to be consumed by code, not by humans, so it's really more of an API. JSON is the common interchange format for APIs.

But also, it's actually kinda more user friendly, in that if we serve `/update.json` with a `application/json` content-type (as we should), most browser will at least display that as text, and some (Firefox) will present it quite nicely. Whereas all of the content-types we could use for `/update.yaml` (`application/x-yaml`, `text/yaml`, `text/x-yaml`) result in the browser prompting the user to download the file, which is much less user friendly.

**Comment by nottrobin** [comment]
I've submitted https://github.com/canonical-web-and-design/multipass.run/pull/64 to actually implement this as an endpoint on the website.

**Comment by ricab** [comment]
Oh dear, I had just finished [moving to yaml](https://github.com/CanonicalLtd/multipass/pull/1212) :sweat: But yeah @nottrobin, good points on the json being preferable for this purpose, I think you're right.

**Pull request merged**